src/NUnitFramework/framework/Constraints/MsgUtils.cs

```diff
@@ -61,6 +61,7 @@ internal static class MsgUtils
         /// Formatting strings used for expected and actual values
         /// </summary>
         private static readonly string Fmt_Null = "null";
+
```

Comment:
Sorry, I did a ReSharper code format on the file without realizing it.  You're right, I should have done that as a separate commit (or not at all).

The blank line in question must have been added by ReSharper.  Looking at the context of the change, it was added because ReSharper seems to want to add blank lines after members which have XMLDoc comments.

Reply:
Let us just keep the improved whitespace changes. Wrt. the XMLDoc, it is also a bit "non-standard" to have a XMLDoc comment that actually covers all the following fields 😄 .

When you add the tests (e.g. just the test from #2518 and a test for regular tuples) can you then also remove this empty line.

Reply:
Where should I add the tests for ValueTuple?  I can't add them to CollectionAssertTest.cs because it uses .NET 2.0.

Reply:
We have at least some options:
1) To include the tests in CollectionAssertTest.cs protecting the new ValueTuple tests with `#if NET_4_5` (similar to what we have done in `MsgUtilTests`).
2) Add a new file that we only include in nunit.framework.tests-4.5.csproj and not in the other .csprojs

Since we are trying to switch to the new csproj format (see #2380) and will have one csproj for each project I think that 1) is the way to go (even though I'm not a fan of conditional directive)

Reply:
ValueTuple was only added to the BCL in .NET Framework 4.7.  Do we need a new project for that framework version?  Or should we add the NuGet package (https://www.nuget.org/packages/System.ValueTuple/4.4.0) to provide support for the 4.5 project?

Reply:
If you add anything, it should only be to the tests. We commonly have had test assemblies at a higher version than what they test. For example, we had (may still have) .NET 3.5 tests of our .NET 2.0 build. We probably don't want a framework build beyond 4.5 until we can use multi-targeting.

Reply:
I agree.  I was asking if we should add a 4.7 test project, or use the NuGet package to provide the ValueTuple type to the 4.5 test project.

Reply:
I'd say the former. But @nunit/framework-team may have other views.

Reply:
Thinking about it some more, I think we would want to use the NuGet package.  It's an official Microsoft package, and is how a lot of people are currently utilizing ValueTuple.  In fact, it's actually how I discovered this problem as I am not targeting 4.7 yet.

We may eventually want to add a 4.7 test project, but I'm not sure it's necessary at the moment for this case.

Reply:
We already use the NuGet package and refer to `System.ValueTuple` from `nunit.framework.tests-4.5.csproj`, so if we follow 1) all we need to do is to add the tests and place the ValueTuple tests inside `#if NET_4_5` (but I'm unsure whether we are talking past each other?)

Reply:
@mikkelbu 's comment makes the most sense to me if we are already including the package.

Reply:
I've been a fan of preprocessor conditions for a long time, but otherwise I agree with Mikkel =D

Reply:
I only wish they were more powerful in C#. I'd love to be able to set a single condition, which then defined other conditions, as well as to use comparison operators. Ah well. 

I guess the future is that such pre-compilation logic will be put into properties in the project file, requiring me to program in XML!

Reply:
I would be okay taking a dependency on the NuGet package, but we will need to add it as a dependency in the `Nuspec` file.

As for the 4.7 target, I've been thinking we may need one soon, but I would prefer to wait until we are using multi-targeting. I have a PR open for that, but I might let that lapse until I get the 3.9 release out.

Reply:
Sorry, I misread, an update to the nuspec isn't necessary. I thought you were adding a dependency on ValueTuple to the framework. Go ahead and add it to the 4.5 tests. I thought it was already there.

Reply:
@CharliePoole you can do some of that type things in t4 templates